metatensor-torch/include/metatensor/torch/labels.hpp

```diff
@@ -63,8 +63,8 @@ class METATENSOR_TORCH_EXPORT LabelsHolder: public torch::CustomClassHolder {
     static TorchLabels empty(torch::IValue names);
 
     /// Create Labels with a single dimension with the given name and values in
-    /// the [0, stop) range
-    static TorchLabels range(std::string name, int64_t end);
+    /// the [0, stop) range, on the given device
+    static TorchLabels range(std::string name, int64_t end, torch::IValue device = torch::IValue(torch::kCPU));
```

Comment:
Ok, fair, I've re-done it using `Labels.to()`. My concern is that it will be a bit slower to create the labels and then move them

Reply:
should be the same cost: we need the labels data on CPU anyway for libmetatensor.so, and the "on-device" version is always a copy. This will just invert the order of the copies.